src/components/container/SurveyTaskList/SurveyTaskList.tsx

```diff
@@ -71,13 +78,53 @@ export default function (props: SurveyTaskListProps) {
 
 	function initialize() {
 
-		var sortIncomplete = function (a: any, b: any) {
-			if (!a.dueDate) { return 1; }
-			if (!b.dueDate) { return -1; }
-			if (parseISO(a.dueDate) > parseISO(b.dueDate)) { return 1; }
-			if (parseISO(a.dueDate) < parseISO(b.dueDate)) { return -1; }
-			return 0;
-		}
+		const sortIncomplete = function (a: SurveyTask, b: SurveyTask): number {
+			const { sortBehavior } = props;
+
+			const type = sortBehavior?.type ?? 'dueDate';
+			const direction = sortBehavior?.direction ?? 'ascending';
+			const sortMultiplier = direction === 'descending' ? -1 : 1;
+
+			switch (type) {
+				case 'alphabetical':
+					return a.surveyName.localeCompare(b.surveyName) * sortMultiplier;
+
+				case 'insertedDate':
+					if (!a.insertedDate) return 1;
```

Comment:
I would add a short circuit condition above this for equality.
```
if (a.insertedDate === b.insertedDate) return 0;
```
Whether they are both `undefined` or both set to the same date string, they should be considered equal by the comparison.